lib/services/local/saml_idp_service_provider.go

```diff
@@ -112,15 +113,9 @@ func (s *SAMLIdPServiceProviderService) GetSAMLIdPServiceProvider(ctx context.Co
 // CreateSAMLIdPServiceProvider creates a new SAML IdP service provider resource.
 func (s *SAMLIdPServiceProviderService) CreateSAMLIdPServiceProvider(ctx context.Context, sp types.SAMLIdPServiceProvider) error {
 	if sp.GetEntityDescriptor() == "" {
-		// fetchAndSetEntityDescriptor is expected to return error if it fails
-		// to fetch a valid entity descriptor.
-		if err := s.fetchAndSetEntityDescriptor(sp); err != nil {
-			s.log.Debugf("Failed to fetch entity descriptor from %q. %v.", sp.GetEntityID(), err)
-			// We aren't interested in checking error type as any occurrence of error mean entity descriptor was not set.
-			if err := s.generateAndSetEntityDescriptor(sp); err != nil {
-				return trace.BadParameter("could not generate entity descriptor with given entity_id %q and acs_url %q: %v",
-					sp.GetEntityID(), sp.GetACSURL(), err)
-			}
+		if err := s.configureEntityDescriptorPerPreset(sp); err != nil {
+			return trace.Wrap(trace.BadParameter("failed to configure entity descriptor with the given entity_id %q and acs_url %q: %v",
+				sp.GetEntityID(), sp.GetACSURL(), err))
```

Comment:
I don't thing that you need to use `trace.Wrap` to additionally wrap the `trace.BadParameter`
```suggestion
			return trace.BadParameter("failed to configure entity descriptor with the given entity_id %q and acs_url %q: %v",
				sp.GetEntityID(), sp.GetACSURL(), err)
```

Reply:
yaa wrapping it below like you suggested in https://github.com/gravitational/teleport/pull/39385#discussion_r1525921803 seems like more proper place.